src/training/flava_eval/glue2.py

```diff
@@ -0,0 +1,324 @@
+        "--seed", type=int, default=0, help="Default random seed."
+    )
+    parser.add_argument(
+        "--separator-token",
+        default="<end_of_text>",
+        type=str,
+        help="Separator token.",
+    )
+    parser.add_argument(
+        "--validation-key",
+        default="validation",
+        type=str,
+        help="Validation key.",
+    )
+    parser.add_argument(
+        "--test-key",
+        default="test",
+        type=str,
+        help="Test key.",
+    )
+
+    args = parser.parse_args(args)
+    return args
+
+
+class EarlyStopping:
+
+    def __init__(self, patience=5, threshold=0.0, metric_name="accuracy"):
+        self.patience = patience
+        self.threshold = threshold
+        self.patience_counter = 0
+        self.best_score = None
+        self.best_metrics = None
+        self.metric_name = metric_name
+
+    def step(self, metrics):
+        score = metrics[self.metric_name]
+        if self.best_score is None:
+            self.best_score = score
+            self.best_metrics = metrics
+        elif score < self.best_score + self.threshold:
+            self.patience_counter += 1
+        else:
+            self.best_score = score
+            self.best_metrics = metrics
+            self.patience_counter = 0
+
+        if self.patience_counter >= self.patience:
+            return True
+        return False
+
+
+class GLUEDataset(Dataset):
+    """GLUE dataset."""
+
+    def __init__(self, task, split, text_key, label_key, separator_token='<end_of_text>', tokenizer=None):
+        super().__init__()
+
+        try:
+            from datasets import load_dataset
+        except ImportError:
+            raise ImportError("Please install HF datasets: pip install datasets")
+        
+        self.dataset = load_dataset("glue", task, split=split)
+        self.label_key = label_key
+        self.text_key = text_key # list of strings
+        self.length = len(self.dataset)
+        self.tokenize = tokenizer
+        self.task = task
+        self.separator_token = separator_token
+
+    def __len__(self):
+        return self.length
+
+    def __getitem__(self, idx):
+        if self.task in MULTI_PROMPT_TASKS:
+            item = self.dataset[idx]
+            text1 = item[self.text_key[0]]
+            text2 = item[self.text_key[1]]
+            label = item[self.label_key]
+            return {
+                'text': self.tokenize([text1 + self.separator_token + text2])[0],
+                'label': label
+            }
+        else:
+            item = self.dataset[idx]
+            text = item[self.text_key[0]]
+            label = item[self.label_key]
+            return {
+                'text': self.tokenize([text])[0],
+                'label': label
+            }
+
+
+class TextClassifier(nn.Module):
+    def __init__(self, encoder, embed_dim, num_labels):
+        super().__init__()
+
+        self.encoder = encoder
+        self.logits_proj = nn.Linear(embed_dim, num_labels)
```

Comment:
Can we verify that it's just a linear projection head instead of an MLP?

Reply:
it is